### PR TITLE
Improve repeated items for slybot

### PR DESCRIPTION
**Pull request by ruairif**
Extract repeated lists of values
Container can only extract a single item unless it wraps a repeated container
Better handle cases where not all repeated items have all fields
Fixed printing error with BaseExtractor
Removed legacy support (legacy now uses scrapely.InstanceBasedLearningExtractor)
Log slybot version when using spidermanager

### CODE REVIEW

File: slybot/slybot/plugins/scrapely_annotations/builder.py [hunk 1]
```diff
@@ -308,12 +309,33 @@ def apply_selector_annotations(annotations, target_page):
...
+                    'id': container_id,
+                    'annotations': {'#portia-content': '#dummy'},
+                    'text-content': '#portia-content',
+                    'container_id': annotation['container_id'],
+                    'field': field,
+                    'tagid': parent.attrib.get('data-tagid')
+                })
+                annotation['item_container'] = True
+                annotation['field'] = field
+                annotation['container_id'] = container_id
```
Comment (sagelliv) [inline slybot/slybot/plugins/scrapely_annotations/builder.py]: Are we mutating the `annotations` that are passed into this function? If so, I think we should separate the mutation from returning the `converted_annotations`.

File: slybot/slybot/plugins/scrapely_annotations/builder.py [hunk 2]
```diff
@@ -308,12 +309,33 @@ def apply_selector_annotations(annotations, target_page):
...
+            continue
+
+        tagids = [int(e.attrib.get('data-tagid', 1e9)) for e in elems]
+        tagid = min(tagids)
+        if tagid is not None:
+            annotation['tagid'] = tagid
+            converted_annotations.append(annotation)
+        if (annotation.get('repeated') and
+                not annotation.get('item_container') and
+                len(annotation.get('annotations')) == 1):
```
Comment (sagelliv) [inline slybot/slybot/plugins/scrapely_annotations/builder.py]: Can you add a local variable describing what this conditional represents? I'm not sure what this case corresponds to so the variable would reveal the intent here.

File: slybot/slybot/plugins/scrapely_annotations/extraction.py [hunk 3]
```diff
@@ -151,6 +151,21 @@ def __init__(self, annotation, attribute_descriptors=None):
     def _create_basic_extractor(cls, annotation, attribute_descriptors):
         return cls(annotation, attribute_descriptors)
 
+    def __str__(self):
+        messages = ['BasicTypeExtractor(']
+        annotation = self.annotation
+        if annotation.surrounds_attribute:
+            messages.append(self.annotation.surrounds_attribute)
```
Comment (sagelliv) [inline slybot/slybot/plugins/scrapely_annotations/extraction.py]: `messages.append(annotation.surrounds_attribute or '')`

File: slybot/slybot/plugins/scrapely_annotations/builder.py (hunk 1 above)
Comment (ruairif) [inline slybot/slybot/plugins/scrapely_annotations/builder.py]: They are meant to be mutated. This function used to just add the `tagid` key to the annotations so they would be applied to the correct elements. Now it also adds support for building a `RepeatedFieldsExtractor` by converting a repeated annotation to a container and a repeated field container.

Here's a quick overview of the 3 different types of containers:
**Container:** `item_container == True`.
**RepeatedContainer:** `item_container == True and repeated == True`.
**RepeatedFieldsContainer:** `item_container == True and repeated == True and field == True`.

**Pull request merged**